jffs2/phoenix-rtos/dev.c

```diff
@@ -114,78 +119,51 @@ jffs2_dev_t *dev_find_ino(void *ptr, unsigned long ino)
 	find.ino = ino;
 
 	mutexLock(dev_common->mutex);
+
 	entry = lib_treeof(jffs2_dev_t, linkage_ino, lib_rbFind(&dev_common->dev_ino, &find.linkage_ino));
 
 	mutexUnlock(dev_common->mutex);
-	return entry;
-}
-
-
-void dev_inc(void *ptr, oid_t *oid)
-{
-	jffs2_dev_t find, *entry;
-	dev_common_t *dev_common = ptr;
-
-	memcpy(&find.dev, oid, sizeof(oid_t));
-
-	mutexLock(dev_common->mutex);
-	entry = lib_treeof(jffs2_dev_t, linkage_oid, lib_rbFind(&dev_common->dev_oid, &find.linkage_oid));
-
-	if (entry != NULL)
-		entry->nlink++;
 
-	mutexUnlock(dev_common->mutex);
+	return entry;
 }
 
 
-void dev_dec(void *ptr, oid_t *oid)
+void dev_destroy(void *ptr, jffs2_dev_t *dev)
```

Comment:
Making unlocked `dev_destroy` version (i.e. `_dev_destroy` that don't take mutex) would allow as to use it in `dev_done` to avoid (slight, but still) code repetition.

Reply:
Done.